lib/chef/resource.rb

```diff
@@ -612,6 +617,20 @@ def run_action(action, notification_type = nil, notifying_resource = nil)
       events.resource_completed(self)
     end
 
+    def with_umask
+      if umask.nil?
+        yield
+        return
+      end
+
+      old_value = ::File.umask(umask.oct)
+      begin
+        yield
+      ensure
+        ::File.umask(old_value)
+      end
+    end
```

Comment:
maybe a bit less wordy:

```
def with_umask
  old_value = ::File.umask
  ::File.umask(umask.oct) unless umask.nil?
  yield
ensure
  ::File.umask(old_value)
end
```

Reply:
I had wanted to avoid touching `File.umask` at all if it didn't need to change, but I think I like your version better and I can add an extra nil check.

Reply:
yeah that would work.